src/shared/components/locales-dropdown/index.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react';
+import { injectIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import localesConfig from '../../../../intl/config.js';
+
+import Link from 'shared/components/link';
+import styles from './index.module.css';
+
+class LocalesDropdown extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isOpened: false,
+        };
+
+        this.availableLocales = localesConfig.availableLocales;
+        this.currentLocale = props.intl.locale;
+
+        this.handleToggleDropdown = this.handleToggleDropdown.bind(this);
+        this.handleOutsideClick = this.handleOutsideClick.bind(this);
+    }
+
+    componentDidMount() {
+        window.addEventListener('click', this.handleOutsideClick);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('click', this.handleOutsideClick);
+    }
```

Comment:
Did you consider using [react-onclickoutside](https://github.com/Pomax/react-onclickoutside) instead of doing your own click outside?

Reply:
I'll consider using it in a future PR.